### PR TITLE
add FilterAction and AgeFilterAction to default taxonomy

### PR DESCRIPTION
**Pull request by cdoron**
Signed-off-by: Doron Chen <cdoron@il.ibm.com>

### CODE REVIEW

File: samples/taxonomy/example/module/actions.yaml [hunk 1]
```diff
@@ -22,6 +24,26 @@ definitions:
...
+    type: object
+    properties:
+      columns:
+        items:
+          type: string
+        type: array
+      age:
+        type: integer
+    required:
+      - columns
```
Comment (shlomitk1) [inline samples/taxonomy/example/module/actions.yaml:46]: Why isn't age required?

File: samples/taxonomy/example/module/actions.yaml [hunk 2]
```diff
@@ -22,6 +24,26 @@ definitions:
...
       - columns
+  FilterAction:
+    type: object
+    properties:
+      columns:
+        items:
+          type: string
+        type: array
+      query:
+        type: string
```
Comment (shlomitk1) [inline samples/taxonomy/example/module/actions.yaml:35]: Columns and query should be required

**Comment by shlomitk1** [comment]
Please add a test (in fybrikapplication_controller_unit_test.go) that checks that the correct arguments are provided in the blueprint. The mockup policy manager should be extended to return the appropriate action.

File: samples/taxonomy/example/module/actions.yaml (hunk 1 above)
Comment (cdoron) [reply samples/taxonomy/example/module/actions.yaml:46]: there is a default value: 18

File: manager/controllers/app/fybrikapplication_controller_unit_test.go [hunk 3]
```diff
@@ -585,6 +585,85 @@ func TestMultipleDatasets(t *testing.T) {
 	g.Expect(plotter.Spec.Flows[1].SubFlows).To(gomega.HaveLen(2))
 }
 
+// XXXXX
```
Comment (shlomitk1) [inline manager/controllers/app/fybrikapplication_controller_unit_test.go:588]: Update the comments

File: manager/controllers/mockup/policymanager.go [hunk 4]
```diff
@@ -112,6 +113,21 @@ func (m *MockPolicyManager) GetPoliciesDecisions(input *policymanager.GetPolicyD
 			respResult = append(respResult, policyManagerResult)
 		}
 
+	case "filter-dataset":
+		actionOnCols := taxonomy.Action{}
+		action := make(map[string]interface{})
+		action[nameKey] = FilterAction
+		filterAction := make(map[string]interface{})
+		filterAction["query"] = "Country == 'United Kingdom'"
```
Comment (shlomitk1) [inline manager/controllers/mockup/policymanager.go:121]: Maybe UK?

File: manager/controllers/app/fybrikapplication_controller_unit_test.go [hunk 5]
```diff
@@ -585,6 +585,91 @@ func TestMultipleDatasets(t *testing.T) {
...
+// Assumptions on response from connectors:
```
Comment (shlomitk1) [inline manager/controllers/app/fybrikapplication_controller_unit_test.go:589]: These and inner comments do not reflect this test

**Review by shlomitk1 — CHANGES_REQUESTED** [review]
Change comments related to the test

File: manager/controllers/app/fybrikapplication_controller_unit_test.go (hunk 5 above)
Comment (cdoron) [reply manager/controllers/app/fybrikapplication_controller_unit_test.go]: fixed

**Review by shlomitk1 — APPROVED** [review]

**Pull request merged**